lib/puppet/type/cisco_fabricpath_global.rb

```diff
@@ -0,0 +1,228 @@
+# Manages the Cisco Fabricpath Global configuration resource.
+#
+# April 2013
+#
+# Copyright (c) 2013-2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+Puppet::Type.newtype(:cisco_fabricpath_global) do
+  @doc = %q(
+    Manages the Cisco Fabricpath Global configuration resource.
+
+    cisco_fabricpath_global {"default":
+      ..attributes..
+    }
+
+    "default" is only acceptable name for this global config object.
+
+    Example:
+    cisco_fabricpath_global { "default":
+      ensure                         => 'present',
+      aggregate_multicast_routes     => false,
+      allocate_delay                 => 10,
+      graceful_merge                 => enable,
+      linkup_delay                   => 10,
+      linkup_delay_enable            => true,
+      linkup_delay_always            => false,
+      loadbalance_multicast_rotate   => 4,
+      loadbalance_multicast_has_vlan => true,
+      loadbalance_unicast_layer      => "mixed",
+      loadbalance_unicast_has_vlan   => true,
+      loadbalance_unicast_rotate     => 10,
+      mode                           => "normal",
+      switch_id                      => 25,
+      transition_delay               => 10,
+      ttl_unicast                    => 32,
+      ttl_multicast                  => 32,
+    }
+  )
+  ###################
+  # Resource Naming #
+  ###################
+
+  # Parse out the title to fill in the attributes in these
+  # patterns. These attributes can be overwritten later.
+  def self.title_patterns
+    identity = ->(x) { x }
+    patterns = []
+
+    # Below pattern matches both parts of the full composite name.
+    patterns << [
+      /^(\S+)$/,
+      [
+        [:name, identity]
+      ],
+    ]
+    patterns
+  end
+
+  newparam(:name, namevar: true) do
+    desc 'ID of the fabricpath global config. Valid values are default.'
+
+    validate do |inst_name|
+      fail "only acceptable name is 'default'" if inst_name != 'default'
+    end
+  end # param id
+
+  ##############
+  # Attributes #
+  ##############
+
+  ensurable
+
+  newproperty(:aggregate_multicast_routes) do
+    desc 'Aggregate Multicast Routes on same tree in the topology.
+          Valid values are true/false'
+    newvalues(:true, :false, :default)
+  end
+
+  newproperty(:allocate_delay) do
+    desc 'Fabricpath Timers Allocate Delay. Valid values are Integer from
+          1..1200 seconds or default'
```

Comment:
We usually try not to enforce range checking because of the differences that can occur on different platforms / releases. Are you reasonably certain that this range will always be the same and will never change for some releases?

Also, this property also supports `'default'`.